streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,271 @@
+        final ClientResponse clientResponse = sendRequest(createTopicsRequest, ApiKeys.CREATE_TOPICS);
+        if (!(clientResponse.responseBody() instanceof CreateTopicsResponse)) {
+            throw new StreamsException("Inconsistent response type for internal topic creation request. Expected CreateTopicsResponse but received " + clientResponse.responseBody().getClass().getName());
+        }
+        final CreateTopicsResponse createTopicsResponse =  (CreateTopicsResponse) clientResponse.responseBody();
+
+        for (InternalTopicConfig internalTopicConfig:topicsMap.keySet()) {
+            short errorCode = createTopicsResponse.errors().get(internalTopicConfig.name()).code();
+            if (errorCode > 0) {
+                if (errorCode == Errors.TOPIC_ALREADY_EXISTS.code()) {
+                    continue;
+                } else {
+                    throw new StreamsException("Could not create topic: " + internalTopicConfig.name() + ". " + createTopicsResponse.errors().get(internalTopicConfig.name()).name());
+                }
+            }
+        }
+    }
+
+    /**
+     * Delets a set of topics.
+     *
+     * @param topics
+     */
+    public void deleteTopics(final Map<InternalTopicConfig, Integer> topics) {
+
+        final Set<String> topicNames = new HashSet<>();
+        for (InternalTopicConfig internalTopicConfig: topics.keySet()) {
+            topicNames.add(internalTopicConfig.name());
+        }
+        deleteTopics(topicNames);
+    }
+
+    /**
+     * Delete a set of topics in one request.
+     *
+     * @param topics
+     */
+    private void deleteTopics(final Set<String> topics) {
+
+        final DeleteTopicsRequest deleteTopicsRequest = new DeleteTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+        final ClientResponse clientResponse = sendRequest(deleteTopicsRequest, ApiKeys.DELETE_TOPICS);
+        if (!(clientResponse.responseBody() instanceof DeleteTopicsResponse)) {
+            throw new StreamsException("Inconsistent response type for internal topic deletion request. Expected DeleteTopicsResponse but received " + clientResponse.responseBody().getClass().getName());
+        }
+        final DeleteTopicsResponse deleteTopicsResponse = (DeleteTopicsResponse) clientResponse.responseBody();
+        for (String topicName: deleteTopicsResponse.errors().keySet()) {
+            if (deleteTopicsResponse.errors().get(topicName).code() > 0) {
+                throw new StreamsException("Could not delete topic: " + topicName);
+            }
+        }
+
+    }
+
+    /**
+     * Send a request to kafka broker of this client. Keep polling until the corresponding response is received.
+     *
+     * @param request
+     * @param apiKeys
+     */
+    private ClientResponse sendRequest(final AbstractRequest request, final ApiKeys apiKeys) {
+
+        String brokerId = null;
+        final SystemTime systemTime = new SystemTime();
```

Comment:
@guozhangwang, yeah, we will have to refactor it to make it more generic, but good to have non-test code using the protocol. :)